src/main/java/io/aiven/connect/jdbc/sink/JdbcSinkConfig.java

```diff
@@ -387,6 +407,17 @@ public JdbcSinkConfig(final Map<?, ?> props) {
         fieldsWhitelist = new HashSet<>(getList(FIELDS_WHITELIST));
         final String dbTimeZone = getString(DB_TIMEZONE_CONFIG);
         timeZone = TimeZone.getTimeZone(ZoneId.of(dbTimeZone));
+        if (pkMode.equals(PrimaryKeyMode.RECORD_KEY)) {
+            // Deletes can be enabled with delete.enabled=true,
+            // but only when the pk.mode is set to record_key.
+            // This is because deleting a row from the table
+            // requires the primary key be used as criteria.
+            deleteEnabled = getBoolean(DELETE_ENABLED);
+        } else {
+            if (getBoolean(DELETE_ENABLED)) {
+                log.error("Delete mode will enabled only if pk mode set to record_key");
+            }
```

Comment:
IMO we should fail here instead of just logging an error. Bonus points if we implement this multi-property validation in the [Connector::validate method](https://kafka.apache.org/37/javadoc/org/apache/kafka/connect/connector/Connector.html#validate(java.util.Map)), but I don't think it's necessary to go that far for this PR if you don't have the time.